### PR TITLE
Merge all cpu in one graph in device overview

### PR DESCRIPTION
**Pull request by Alucardfh**
Hi,

This provide a way to hide all separate graphs in the device overview and replace it by the average cpu usage.

This make the overview easier to read and render the page a bit faster as there is less graphs to draw.

This is an optional feature, disabled by default.

Just add in your config.php : 

$config['cpu_details_overview'] = false;

This is related to issue #2218

cpuovreview

cpuovreviewchange

### CODE REVIEW

**Comment by laf** [comment]
I've not checked any code yet but can you update this to have it ON by default. It's horrible so I don't expect people will miss it. It makes the UI better this way.

**Comment by laf** [comment]
Had a chance to check through this, some comments below and also inline with your code:

 - I'd say we have this on by default and people can turn it off.

 - The code formatting needs to adhere to the structure we have here: http://docs.librenms.org/Developing/Code-Guidelines/

File: html/pages/device/overview/processors.inc.php [hunk 1]
```diff
@@ -16,44 +16,88 @@
...
+    $totalPercent=0;
+
     foreach ($processors as $proc) {
         $text_descr = rewrite_entity_descr($proc['processor_descr']);
 
         // disable short hrDeviceDescr. need to make this prettier.
         // $text_descr = short_hrDeviceDescr($proc['processor_descr']);
         $percent      = $proc['processor_usage'];
-        $background   = get_percentage_colours($percent);
-        $graph_colour = str_replace('#', '', $row_colour);
```
Comment (laf) [inline html/pages/device/overview/processors.inc.php]: Should this line and the above be moved? Are these variables not used in both parts of the if / else block?

File: html/pages/device/overview/processors.inc.php [hunk 2]
```diff
@@ -16,44 +16,88 @@
...
+            $graph_colour = str_replace('#', '', $row_colour);
+
+            $graph_array           = array();
+            $graph_array['height'] = '100';
+            $graph_array['width']  = '210';
+            $graph_array['to']     = $config['time']['now'];
+            $graph_array['id']     = $proc['processor_id'];
+            $graph_array['type']   = $graph_type;
+            $graph_array['from']   = $config['time']['day'];
+            $graph_array['legend'] = 'no';
```
Comment (laf) [inline html/pages/device/overview/processors.inc.php]: We now have a lot of duplicated variables which could just be set before the if check.  Can you consolidate them together and then just overwrite where you need.

File: html/pages/device/overview/processors.inc.php [hunk 3]
```diff
@@ -16,44 +16,88 @@
...
+              </tr>';
+        }
+        else {
+            $totalPercent = $totalPercent + $percent;
+        }
 
-        $graph_array           = array();
+    }//end foreach
+
+    if ($config['cpu_details_overview'] !== true)
```
Comment (laf) [inline html/pages/device/overview/processors.inc.php]: Can you not change this to be === false?

File: html/pages/device/overview/processors.inc.php [hunk 4]
```diff
@@ -16,44 +16,88 @@
...
+    {
         $graph_array['height'] = '100';
-        $graph_array['width']  = '210';
+        $graph_array['width']  = '485';
```
Comment (laf) [inline html/pages/device/overview/processors.inc.php]: What's the reason for changing the width?

File: html/pages/device/overview/processors.inc.php [hunk 5]
```diff
@@ -16,44 +16,88 @@
...
         $graph_array['to']     = $config['time']['now'];
-        $graph_array['id']     = $proc['processor_id'];
-        $graph_array['type']   = $graph_type;
+        $graph_array['device'] = $device['device_id'];
+        $graph_array['type']   = 'device_processor';
```
Comment (laf) [inline html/pages/device/overview/processors.inc.php]: Are these changes necessary?

File: html/pages/device/overview/processors.inc.php [hunk 6]
```diff
@@ -16,44 +16,88 @@
...
         $link = generate_url($link_array);
 
-        $overlib_content = generate_overlib_content($graph_array, $device['hostname'].' - '.$text_descr);
-
-        $graph_array['width']  = 80;
-        $graph_array['height'] = 20;
-        $graph_array['bg']     = 'ffffff00';
-        // the 00 at the end makes the area transparent.
-        $minigraph =  generate_lazy_graph_tag($graph_array);
+        $graph_array['width'] = '210';
```
Comment (laf) [inline html/pages/device/overview/processors.inc.php]: Width has been changed, what's the reason for that?

File: html/pages/device/overview/processors.inc.php (hunk 4 above)
Comment (Alucardfh) [inline html/pages/device/overview/processors.inc.php]: The width change is to make the graph of a normal size instead of the mini graph when all cpu are displayed.

File: html/pages/device/overview/processors.inc.php (hunk 5 above)
Comment (Alucardfh) [inline html/pages/device/overview/processors.inc.php]: Yes as it is not the same graph type, the old one use "processor_usage" which works with the id of the cpus but the new way works with the device ID and the "device_processor" graph type.

File: html/pages/device/overview/processors.inc.php (hunk 2 above)
Comment (Alucardfh) [inline html/pages/device/overview/processors.inc.php]: I agree, it would be cleaner to have less duplicate, I'll see to consolidate everything above and just override what is needed.

File: html/pages/device/overview/processors.inc.php (hunk 1 above)
Comment (Alucardfh) [inline html/pages/device/overview/processors.inc.php]: Only the $background is used in both part, as actualy $graph_colour doesn't seem to be used.

But it make sense to compute the background colour on the final precentage and not on the last one.
I'll move this to the else block.

I need to clean up the variables that are not used.

**Comment by Alucardfh** [comment]
I'll work on the comments when I get time (probably Monday).

**Comment by laf** [comment]
No problem.  Thanks for the work :)

**Comment by Alucardfh** [comment]
I have reorganized the code and trimmed what wasn't used.

I also switch it on by default.
